daemon/src/wallpaper_config.rs

```diff
@@ -52,4 +52,9 @@ Either remove `duration` or set `path` to a directory"
     pub fn get_output_by_name(&self, name: &str) -> Arc<WallpaperInfo> {
         self.data.get(name).unwrap_or(&self.default_config).clone()
     }
+
+    /// Return true only if the data inside the wallpaper config is not as same as the right one
+    pub fn is_same(&self, rhs: &Self) -> bool {
```

Comment:
I think this function should be replaced by the trait [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html). We already derive the trait but as you pointed out a custom implementation is better here.

Reply:
Oh yes, I forgot the PartialEq trait.